code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingImageDownloaderTask.java

```diff
@@ -44,30 +48,43 @@ class MessagingImageDownloaderTask implements Callable<Bitmap> {
      */
     @Override
     public Bitmap call() {
-        return download(url);
+        return download();
     }
 
     /**
-     * Download the image asset and converts it to a {@link Bitmap}.
+     * Download the image asset using the {@code RemoteDownloader} and convert it to a {@code Bitmap}. Downloaded image assets are
+     * stored in the Message extension cache directory. A {@code CountDownLatch} is used to make the download synchronous as the downloaded
+     * image will be used immediately in a displayed push notification.
      *
-     * @param url {@code String} containing the image asset url to be downloaded
      * @return the downloaded image asset as a {@link Bitmap}
      */
-    private Bitmap download(final String url) {
-        Bitmap bitmap = null;
-        HttpURLConnection connection = null;
+    private Bitmap download() {
+        RemoteDownloader remoteDownloader;
+        final Bitmap[] bitmap = new Bitmap[1];
+        final CountDownLatch latch = new CountDownLatch(1);
         try {
-            final URL imageUrl = new URL(url);
-            connection = (HttpURLConnection) imageUrl.openConnection();
-            bitmap = BitmapFactory.decodeStream(connection.getInputStream());
-        } catch (final IOException e) {
-            Log.warning(LOG_TAG, "%s - Exception occurred when downloading the image: %s", SELF_TAG, e.getMessage());
-        } finally {
-            if (connection != null) {
-                connection.disconnect();
-            }
+            remoteDownloader = new RemoteDownloader(networkService, systemInfoService, url, IMAGES_CACHE_SUBDIRECTORY) {
+                @Override
+                protected void onDownloadComplete(final File downloadedFile) {
+                    if (downloadedFile != null) {
+                        bitmap[0] = MessagingUtils.getBitmapFromFile(downloadedFile);
+                        latch.countDown();
+                    } else {
+                        Log.debug(LOG_TAG, "%s - Failed to download asset from (%s).", SELF_TAG, url);
+                    }
+                }
+            };
+        } catch (final MissingPlatformServicesException exception) {
+            Log.warning(LOG_TAG, "%s - Failed to download the image asset: (%s), the platform services were not available.", SELF_TAG, url);
+            return null;
         }
-
-        return bitmap;
+        remoteDownloader.startDownload();
+        try {
+            latch.await(2000, TimeUnit.MILLISECONDS);
```

Comment:
google recommends that messages without data that needs to be processed should be handled within 10 seconds. this will be removed though as I am now using startDownloadSync.